openfl/experimental/workflow/runtime/federated_runtime.py

```diff
@@ -2,71 +2,109 @@
 # SPDX-License-Identifier: Apache-2.0
 
 
-"""openfl.experimental.workflow.runtime package LocalRuntime class."""
+"""openfl.experimental.workflow.runtime package FederatedRuntime class."""
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+import logging
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple
 
-from openfl.experimental.workflow.runtime.runtime import Runtime
+import dill
+from tabulate import tabulate
 
-if TYPE_CHECKING:
-    from openfl.experimental.workflow.interface import Aggregator, Collaborator
+from openfl.experimental.workflow.runtime.runtime import Runtime
+from openfl.experimental.workflow.transport.grpc.director_client import DirectorClient
+from openfl.experimental.workflow.workspace_export import WorkspaceExport
 
-from typing import List, Type
+logger = logging.getLogger(__name__)
 
 
 class FederatedRuntime(Runtime):
-    """Class for a federated runtime, derived from the Runtime class.
+    """FederatedRuntime class, derived from Runtime class.
 
     Attributes:
-        aggregator (Type[Aggregator]): The aggregator participant.
-        collaborators (List[Type[Collaborator]]): The list of collaborator
-            participants.
+        _aggregator (Optional[str]): Name of the aggregator. Defaults to
+                None.
+        __collaborators (Optional[List[str]]): List of Authorized collaborators
+        tls (bool): A flag indicating if TLS should be used for
+            connections. Defaults to False.
+        director (Optional[Dict[str, Any]]): Dictionary containing director info.
+        _dir_client (DirectorClient): The director client.
+        notebook_path (Optional[str]) : Path to the Jupyter notebook
+        experiment_submitted (bool): Whether the experiment has been submitted.
+        generated_workspace_path (Path): Path to generated workspace
     """
 
     def __init__(
         self,
-        aggregator: str = None,
-        collaborators: List[str] = None,
-        **kwargs,
+        aggregator: Optional[str] = None,
+        collaborators: Optional[List[str]] = None,
```

Comment:
Is it possible to remove the `aggregator` and `collaborator` objects from here? From an abstract point of view, the `FederatedRuntime` should only know about the Director, who manages the aggregator (and collaborators via the envoys).